feat: Add a bunch o' new panels to the homepage.

sarahmonster opened the pull request:
Needs fleshing out & fine-tuning, obviously, but this is just about getting the new structure in place.

This is what it should (sort of) look like when it's finished:
<img src="https://user-images.githubusercontent.com/376315/72175715-b5899f00-33d4-11ea-9f01-9c13c5d053cb.png" width="320" alt="Mockup of octopusthink.com homepage, showing lots of different panels of content." />

Notes & commentary on #105.

@tofumatt most of this work is going to be a bunch of CSSing/design work that I can handle, but I'd like your help with three things:

- [x] Pulling the two most recent blog posts into the blog panel.
- [x] Showing three portfolio items in the portfolio panel. (Ideally this would be either random selection, or numbers defined somewhere in the homepage MDX. Failing those options, using portfolio item frontmatter is acceptable.
- [ ] Copyedits & content-thoughts for the final panel ("Work with us"). 

I'll be asking for more fine-grained review as I pull everything together, but those are the salient points here.

tofumatt commented:
I've added the portfolio and the blog components that pull in dynamic data to the homepage. I've not done the copy-edits yet, but I think the blog posts bit will need some styling 😄

sarahmonster commented:
Did some styling tweaks! Still could use refining:

- [ ] consider one CTA (rather than two) for blog panel
- [ ] accessibility check
- [ ] fine-tune logos on smaller devices (requires adding mobileColumns/tabletColumns/desktopTablets etc props to GridLayout, like panel)
- [ ] hide second portfolio item on tablets
- [ ] shorten services text on mobile
- [ ] replace about photo
- [ ] replace constructopus with an illustration
- [ ] copyedit "work with us" text

sarahmonster commented:
I rebased and force-pushed, don't hate me!

I've fixed as many of the accessibility issues flagged by Axe as I'm able to/can figure out, but I'm not certain about some of them. 

What's left for me to do:
- Fine-tuning the pinks & cyans to ensure they have sufficient contrast (finicky, but in progress)
- Ensure portfolio items have some kind of focus state (can be done as an add-on, I suppose)

I'm having to manually edit out ids from svgs, so we definitely want svgo (#48) at some point, espeically if we want to do more svg work.

Anyway. I think this is probably about ready to review/merge now.

tofumatt commented on src/components/FriendLogos/index.js:
```diff
@@ -3,37 +3,41 @@ import React from 'react';
+        {!abbreviated && <Image src="wordpress.com.png" alt="WordPress.com logo" />}
         <Image src="snp.png" alt="Scottish National Party logo" />
-        <Image src="ns-government.png" alt="Nova Scotia Government logo" />
+        <Hide mobile>
```
What is happening here? Why is this hidden on mobile at all times? Should this only be hidden when in abbreviated mode or on both pages? This seems a wee bit fragile, and just want to double-check the intention here.

tofumatt replied:
I figured out what's happened by running the code and this causes an uneven number of items on mobile on the _main_ portfolio page:

<img width="372" alt="Screenshot 2020-01-23 19 38 31" src="https://user-images.githubusercontent.com/90871/73017598-09a27380-3e18-11ea-9234-ac75b7148639.png">

It seems what you meant was `!abbreviated ? <Image src="ns-government.png" alt="Nova Scotia Government logo" /> : (<Hide mobile><Image src="ns-government.png" alt="Nova Scotia Government logo" /></Hide>)`

I don't love it, but it'll do. I've made the change.
